msteams-platform/m365-apps/build-actions-in-m365.md

```diff
@@ -0,0 +1,165 @@
+        "id": "relatedTasks",
+        "displayName": "Related tasks",
+        "intent": "custom",
+        "description": "Shows tasks in the To do app that are related to this file.",
+        "handlers": [
+            {
+                "type": "openPage",
+                "supportedObjects": {
+                    "file": {
+                        "extensions": ["xlsx", "doc", "docx", "pdf", "pptx", "ppt"]
+                    }
+                },
+                "pageInfo": {
+                    "pageId": "index",
+                }
+            }
+        ]
+    }
+]
+}
+```
+
+For more information, see [public developer preview app manifest schema](../resources/schema/manifest-schema-dev-preview.md#actions).
+
+When a user selects an action to open a page and view related tasks in an app based on the selected file. The app uses the `"intent": "custom"` property to identify the file type, such as .xlsx or doc and the `"type": "openPage"` handler opens the app and navigates to the `pageId`.
+
+## Retrieve Action information through context object
+
+Update the handler to receive the Action information through the context object to create a seamless user experience for performing users specific tasks using the [Teams JavaScript library (TeamsJS)](/javascript/api/@microsoft/teams-js).
+
+When a user selects Add option from the app's context menu, a page opens with the help of the `openPage` property in the app manifest. Your app can access contextual information about the invoked Action from the `actionInfo` property of the context object `app.getContext()`.
+
+```javascript
+const context = await app.getContext();
+const itemId = context.actionInfo && context.actionInfo.actionObjects[0].itemId;
+this.setState({
+    itemId: itemId
+});
+```
+
+| &nbsp; | Name | Description |
+| --- | --- | --- |
+| &nbsp; | `actionId` | Maps to the Action ID supplied inside the manifest. |
+| &nbsp; | `actionObjects` | Array of corresponding action objects. |
+| &nbsp; | `itemId` | The app receives the ID as the content and uses it to query the Microsoft graph. |
+
+The [ActionInfo](/javascript/api/@microsoft/teams-js/actioninfo) interface helps to enable your app to determine when a user opens a page from an Action and the content that initiated the Action.
+
+```javascript
+app.getContext().then((context: app.Context) => {
+    const actionInfo = context.actionInfo;
+    if (actionInfo && actionInfo.actionId == 'myActionId1') {
+        // Handle specific action
+    }
+});
+```
+
+## Access content through Graph API
+
+After obtaining the `itemId` of the triggering content, you can use the Graph API to read or modify the content, facilitating task completion for your users.
+
+```javascript
+async readActionItem() {
+    try {
+        return await this.graphClient.api(`/users/${this.objectId}/drive/items/${this.itemId}`).get();
+    } catch (error) {
+        console.log("readActionItem", error);
+    }
+}
+```
+
+## Run your app
+
```

Comment:
You missed the part to make sure your account is in Targeted Release. Please refer to the original doc: 

Enroll your account into Targeted Release 

For this preview, only targeted release users will be able to experience Actions in Microsoft 365 app. Make sure your account is in Targeted Release before trying it out. To enroll in Targeted release, please refer to [Set up the Standard or Targeted release options - Microsoft 365 admin | Microsoft Learn](https://learn.microsoft.com/en-us/microsoft-365/admin/manage/release-options-in-office-365?view=o365-worldwide&preserve-view=true#targeted-release)